packages/itwin/tree-widget/src/components/trees/repositories-tree/RepositoriesTree.tsx

```diff
@@ -0,0 +1,81 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+import { useState } from "react";
+import { Flex, ProgressRadial, Text } from "@itwin/itwinui-react";
+import { TreeRenderer, UnifiedSelectionProvider, useUnifiedSelectionTree } from "@itwin/presentation-hierarchies-react";
+import { TreeWidget } from "../../../TreeWidget";
+import { Delayed } from "../common/components/Delayed";
+import { ProgressOverlay } from "../common/components/ProgressOverlay";
+import { getRepositoryNodeIcon } from "./GetIcon";
+import { UseRepositoriesHierarchyProvider } from "./UseRepositoriesHierarchyProvider";
+
+import type { SelectionStorage } from "@itwin/presentation-hierarchies-react";
+
+interface RepositoriesTreeProps {
+  selectionStorage: SelectionStorage;
+  accessToken: string;
+  itwinId: string;
+  environment?: "PROD" | "QA" | "DEV";
+  noDataMessage?: string;
+}
+
+/**
+ * @alpha
+ */
+export function RepositoriesTree({ selectionStorage, ...props }: RepositoriesTreeProps) {
+  return (
+    <UnifiedSelectionProvider storage={selectionStorage}>
+      <Tree {...props} />
+    </UnifiedSelectionProvider>
+  );
+}
+
+function Tree({ accessToken, itwinId, noDataMessage, environment }: Omit<RepositoriesTreeProps, "selectionStorage">) {
+  const [isLoading, setIsLoading] = useState(true);
```

Comment:
I don't think we need to handle this ourselves. `useTree` hooks return an `isLoading` flag which we can use.

Reply:
changed